src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs

```diff
@@ -22,7 +22,9 @@ public static partial class JsonSerializer
             }
 
             // Determine if we are done enumerating properties.
-            if (state.Current.PropertyIndex != classInfo.PropertyCount)
+            // If the ClassType is unknown, there will be a policy property applied. There is probably
+            // a better way to identify policy properties- maybe not put them in the normal property bag?
+            if (classInfo.ClassType != ClassType.Unknown && state.Current.PropertyIndex != classInfo.PropertyCount)
```

Comment:
Fwiw the original ideas of ClassType.Unknown was to convert it to the "real" type via obj.GetType() (either as a value or object) during deserialize and not keep it as unknown.